### DISCUSSION TITLE
Gitlab Homebrew Formula publish returns 404

### DISCUSSION THREAD

**Discussion by bitmasksn** [goreleaser/goreleaser#2898] [Q&A] [answered]
On a self-hosted Gitlab, with the following in .goreleaser.yaml:
```
brews:
  -
    name: myproject
    tap:
      name: 44
      branch: main

      token: "{{ .Env.HOMEBREW_TAP_GITLAB_TOKEN }}"
```
I get the following error in the pipeline:

```
      • homebrew tap formula
         • pushing                   formula=myproject.rb repo=/44
         ⨯ error creating brew formula file branch=main err=POST https://gitlab.mycompany.com/api/v4/projects//44/repository/files/myproject.rb: 404 {message: 404 Project Not Found} fileName=myproject.rb projectID=/44 statusCode=404
   ⨯ release failed after 29.16s error=homebrew tap formula: failed to publish artifacts: POST https://gitlab.mycompany.com/api/v4/projects//44/repository/files/myproject.rb: 404 {message: 404 Project Not Found}
```
The preceding creation of the release and the push of the release artifacts that the formula will point to all succeed, and are defined in the same .goreleaser.yaml, use the same token and push to the same repository. 

Also, I took the url from the error shown and copy/pasted it into a curl, using the same token defined in `HOMEBREW_TAP_GITLAB_TOKEN`. POSTed it to the Gitlab API and it works fine.

Any ideas on what might be happening? Any thoughts on how I can debug this further... maybe there is a -debug or -verbose mode?

**Comment by caarlos0** [comment, marked as answer]
same as https://github.com/goreleaser/goreleaser/issues/2954  ?

**Reply by bitmasksn** [reply]
Yes, it appears so. Because of this issue I abandoned using goreleaser to create the formula, instead I create it myself so I can use the project id in the gitlab project path as referenced in https://github.com/goreleaser/goreleaser/issues/2954#issuecomment-1060037829. So I suspect that a fix for [#2954](https://github.com/goreleaser/goreleaser/issues/2954) will resolve this issue as well

**Reply by caarlos0** [reply]
ok, will mark this as answer then, as it'll be handled in the issue
